src/edgepi/pwm/edgepi_pwm.py

```diff
@@ -175,18 +180,20 @@ def close(self, pwm_num: PWMPins):
         Returns:
             N/A
         """
-        if pwm_num is None or pwm_num not in self.__pwm_devs:
-            raise ValueError(f"get_enabled: PWM number is missing {pwm_num}")
-        self.__pwm_devs[pwm_num].close_pwm()
-        self.__pwm_devs[pwm_num] = None
+        if pwm_num is None or pwm_num not in EdgePiPWM.__pwm_devs:
+            raise ValueError(f"close: PWM number is missing {pwm_num}")
+        with EdgePiPWM.__lock_pwm[pwm_num]:
+            EdgePiPWM.__pwm_devs[pwm_num].close_pwm()
+            EdgePiPWM.__pwm_devs[pwm_num] = None
```

Comment:
if `EdgePiPWM.__pwm_devs[pwm_num] = None` has to be done on close, I think we should wrap `close_pwm` in try-catch-finally as well. something like:

```
        with EdgePiPWM.__lock_pwm[pwm_num]:
            try:
                EdgePiPWM.__pwm_devs[pwm_num].close_pwm()
            except:
                raise
            finally:
                 EdgePiPWM.__pwm_devs[pwm_num] = None
```

Reply:
[7725deb](https://github.com/EdgePi-Cloud/edgepi-python-sdk/pull/400/commits/7725deb4c20419d13af8a8e01cf95da3c8105d49)